Realm/Tests/RealmTests.mm

```diff
@@ -1158,4 +1158,22 @@ - (void)testNotificationPipeBufferOverfull {
     }
 }
 
+- (void)testHoldRealmAfterSourceThreadIsDestroyed {
+    __block RLMRealm *realm;
+
+    // Using an NSThread to ensure the thread (and thus runloop) is actually destroyed
+    NSThread *thread = [[NSThread alloc] initWithTarget:self selector:@selector(runBlock:) object:^{
+        realm = [RLMRealm defaultRealm];
+    }];
+    [thread start];
+    while (!thread.isFinished)
+        usleep(100);
+
+    [realm path]; // ensure ARC releases the object after the thread has finished
```

Comment:
How does this ensure the object is released?

Could we not use a __weak variable to check this?